<?xml version="1.0"?>
<launch>
    <arg name="world_file" default="hospital.sdf"/>
    <arg name="config_file" default="semantic_segmentation_gui.config"/>

    <include file="$(find ros_ign_gazebo)/launch/ign_gazebo.launch">
        <arg name="ign_args" value="-r -v 1 $(find semantic_gazebo_simulation)/worlds/$(arg world_file) --gui-config $(find semantic_gazebo_simulation)/worlds/gui_config/$(arg config_file)"/>
    </include>

    <node pkg="semantic_gazebo_simulation" type="camera_pose_transformer" name="camera_pose_transformer" output="screen"/>

    <node pkg="ros_ign_bridge" type="parameter_bridge" name="$(anon ros_ign_bridge)" output="screen" args="/rgbd_camera/camera_info@sensor_msgs/CameraInfo@ignition.msgs.CameraInfo 
                /rgbd_camera/image@sensor_msgs/Image@ignition.msgs.Image 
                /rgbd_camera/depth_image@sensor_msgs/Image@ignition.msgs.Image 
                /rgbd_camera/points@sensor_msgs/PointCloud2@ignition.msgs.PointCloudPacked 
                /semantic/colored_map@sensor_msgs/Image@ignition.msgs.Image 
                /semantic/labels_map@sensor_msgs/Image@ignition.msgs.Image 
                /model/semantic_rgbd_camera/pose@geometry_msgs/PoseStamped@ignition.msgs.Pose
                /model/semantic_rgbd_camera/cmd_vel@geometry_msgs/Twist@ignition.msgs.Twist"/>
    
    <rosparam file="$(find semantic_gazebo_simulation)/config/label_colors.yaml" command="load"/>
    <node pkg="semantic_gazebo_simulation" type="semantic_cloud_publisher.py" name="semantic_cloud_publisher" output="screen"/>


    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find semantic_gazebo_simulation)/rviz/rviz_semantic_gazebo_simulation.rviz"/>
    
    <node pkg="tf" type="static_transform_publisher" name="robot_start_frame" args="6 3 2 1.5707 0 0 semantic_segmentation_world robot_start_frame 0" />
    
    <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_twist_keyboard" output="screen" launch-prefix="xterm -e">
        <remap from="cmd_vel" to="/model/semantic_rgbd_camera/cmd_vel"/>
    </node>


</launch>
